lib/processAllArtifacts.js

```diff
@@ -12,14 +13,59 @@ var readFile = util.promisify(fs.readFile)
 var writeFile = util.promisify(fs.writeFile)
 var readdir = util.promisify(fs.readdir)
 
-module.exports = async function (inputArtifactsPath, outputArtifactsPath, networksPath) {
-  let inputDirError = false
+const updateArtifactFiles = (inputArtifactsPath, outputArtifactsPath, networkConfigs) => {
+  return new Promise(async (resolve, reject) => {
+    await readdir(inputArtifactsPath).then(async filenames => {
+      await Promise.all(filenames.map(async filename => {
+        const filepath = path.join(inputArtifactsPath, filename)
+        const json = fs.readFileSync(filepath)
+        const artifact = JSON.parse(json)
+
+        const updatedArtifact = await updateArtifact(artifact, networkConfigs)
+
+        const artifactOutputPath = path.join(outputArtifactsPath, filename)
+        const outputJSON = JSON.stringify(updatedArtifact, null, 2)
+        fs.writeFileSync(artifactOutputPath, outputJSON)
+
+        console.log(chalk.cyan(`Updated existing artifact: ${filename}`))
+      }))
+
+      console.log(chalk.grey('Updating complete!'))
+      resolve()
+    })
+  })
+}
+
+const createMissingArtifactFiles = (outputArtifactsPath, networkConfigs) => {
```

Comment:
Likewise here, just make it async instead of making a promise.  Less code and it might make error handling cleaner.